src/v/cluster/controller_backend.cc

```diff
@@ -455,7 +458,18 @@ controller_backend::bootstrap_ntp(const model::ntp& ntp, deltas_t& deltas) {
     }
     // apply all deltas following the one found previously
     deltas = std::move(bootstrap_deltas);
-    return reconcile_ntp(deltas);
+    return reconcile_ntp(deltas).then([this, deltas] {
+        if (deltas.size() == 0) {
+            return ss::make_ready_future<void>();
+        }
+        auto last_delta = deltas.back();
+        if (last_delta.type == op_t::del) {
+            return cleanup_orphan_files(
```

Comment:
have you considered doing this in log_manager::remove()? Some where around. 

```
  return lg.remove()
          .then([dir = std::move(ntp_dir)] { return ss::remove_file(dir); }) <===
```


All the parsing logic from ntp seems a bit out of place here. Its possible I could be missing something here, so just wanted to double check.

Reply:
i think this should be part of processing partition deletion update in `delete_partition`. This `bootstrap_ntp` method doesn't directly cause any side effects

Reply:
moved to reconciliation loop